### ISSUE TITLE
Add a way to make an @Inject optional

### ISSUE THREAD

**Issue by Runemoro** [SpongePowered/Mixin#297]
Currently, Mixin throws an exception if the target method isn't found:

https://github.com/SpongePowered/Mixin/blob/master/src/main/java/org/spongepowered/asm/mixin/injection/struct/InjectionInfo.java#L419-L424

There should be a way to tell Mixin that the target method may not exist at runtime and not throw the exception, for example if:
 - Another mod removes the Vanilla method you're targeting (Optifine seems to be doing this somehow), and the Mixin isn't that important
 - You're targeting a method in another mod's class to make it compatible with your mod, and you'd like the Mixin to just not be applied rather than crashing the whole game if the author deletes the method in a future version of the mod

**Comment by DemonWav** [comment]
Shouldn't you use Mixin plugins for this?

**Comment by DemonWav** [comment]
Or not even that, just an optional Mixin config: https://github.com/SpongePowered/Mixin/blob/master/src/main/java/org/spongepowered/asm/mixin/extensibility/IMixinConfig.java#L77-L83

See example: https://github.com/SpongePowered/SpongeCommon/blob/stable-7/src/main/resources/mixins.common.core.optional.json

**Comment by Runemoro** [comment]
> Shouldn't you use Mixin plugins for this?

That's what I ended up doing for Optifine. But the mixin really *is* an optional feature, and I'd like a way to specify that I'd prefer the mixin not to be applied if the method is missing (for any reason, not just Optifine).

For the second case (mixing into a mod's class), I can't (easily) use a mixin plugin since I don't know in what future version of the mod the author might move or delete the method. If I did use a mixin plugin, I'd have to get the mixin target's class bytes and disable my mixin if the class doesn't contain the method.

> Or not even that, just an optional Mixin config: 

I tried that. That disables the exception if the injection point in the method is not found, not if the whole method is not found.
